protobuf/lib/src/protobuf/proto3_json.dart

```diff
@@ -0,0 +1,326 @@
+          throw parseException(
+              'Expected a double represented as a String or number');
+        case PbFieldType._ENUM_BIT:
+          if (value is String) {
+            // TODO(sigurdm): Do we want to avoid linear search here? Measure...
+            return fieldInfo.enumValues.firstWhere((e) => e.name == value,
+                orElse: () =>
+                    throw parseException('Unknown enum value: $value'));
+          } else if (value is int) {
+            return fieldInfo.valueOf(value) ??
+                (throw parseException('Unknown enum value: $value'));
+          }
+          throw parseException('Expected enum as a string or integer');
+        case PbFieldType._INT32_BIT:
+        case PbFieldType._SINT32_BIT:
+        case PbFieldType._UINT32_BIT:
+        case PbFieldType._FIXED32_BIT:
+        case PbFieldType._SFIXED32_BIT:
+          if (value is int) return value;
+          if (value is String) return int.parse(value);
+          throw parseException('Expected int or stringified int');
+          break;
+        case PbFieldType._INT64_BIT:
+        case PbFieldType._SINT64_BIT:
+        case PbFieldType._UINT64_BIT:
+        case PbFieldType._FIXED64_BIT:
+        case PbFieldType._SFIXED64_BIT:
+          if (value is int) return Int64(value);
+          if (value is String) return Int64.parseInt(value);
+          throw parseException('Expected int or stringified int');
+        case PbFieldType._GROUP_BIT:
+        case PbFieldType._MESSAGE_BIT:
+          GeneratedMessage subMessage = fieldInfo.subBuilder();
+
+          helper(value, subMessage._fieldSet);
+          return subMessage;
+        default:
+          throw StateError('Unknown type $fieldType');
+      }
+    }
+
+    Object decodeMapKey(String key, int fieldType) {
+      switch (PbFieldType._baseType(fieldType)) {
+        case PbFieldType._BOOL_BIT:
+          switch (key) {
+            case 'true':
+              return true;
+            case 'false':
+              return false;
+            default:
+              throw parseException(
+                  'Wrong boolean key "$key", should be one of ("true", "false")');
+          }
+          // ignore: dead_code
+          throw StateError('(Should have been) unreachable statement');
+        case PbFieldType._STRING_BIT:
+          return key;
+        case PbFieldType._UINT64_BIT:
+          // TODO(sigurdm): unsigned...
+          return Int64.parseInt(key);
+        case PbFieldType._INT64_BIT:
+        case PbFieldType._SINT64_BIT:
+        case PbFieldType._SFIXED64_BIT:
+        case PbFieldType._FIXED64_BIT:
+          return Int64.parseInt(key);
+        case PbFieldType._INT32_BIT:
+        case PbFieldType._SINT32_BIT:
+        case PbFieldType._FIXED32_BIT:
+        case PbFieldType._SFIXED32_BIT:
+          // TODO(sigurdm): Restrict to 32 bits.
+          return int.parse(key);
+        case PbFieldType._UINT32_BIT:
+          // TODO(sigurdm): unsigned...
```

Comment:
?

Reply:
I made better handling.
Leaving unsigned 64 bit numbers unchecked for now...